JPS_BASE_LIB/src/test/java/uk/ac/cam/cares/jps/base/annotate/test/MetaDataQueryTest.java

```diff
@@ -0,0 +1,289 @@
+                "testToCreationTime", "testIriCreatingAgent","testFromSimulationTime",
+                "testToSimulationTime","testIriScenario", Arrays.asList("topic1","topic2","topic3"));
+
+        Assert.assertEquals(expected_nullArg,actual_nullArg);
+        Assert.assertEquals(expected,actual);
+
+    }
+
+    @Test
+        public void testQuery1(){
+            String sparql="test";
+            String expected="queryResultString";
+            try (MockedStatic<AccessAgentCaller> aac= Mockito.mockStatic(AccessAgentCaller.class)){
+                aac.when(()->AccessAgentCaller.query(MetaDataAnnotator.getMetadataSetUrl(),null,sparql)).thenReturn(expected);
+                String actual=MetaDataQuery.query(sparql);
+                Assert.assertEquals(expected,actual);
+            }
+        }
+
+    @Test
+        public  void testQuery2(){
+        //Two cases are distinguished. One with an empty metadataseturl and one which is not
+
+        //The case with a non null metadataseturl also has to be mocked because if the server is
+        //inaccessible an HTTP exception is thrown
+        String sparql="PREFIX dcterms:<http://purl.org/dc/terms/> \r\n"
+                +"PREFIX xsd:<http://www.w3.org/2001/XMLSchema#> \r\n"
+                +"PREFIX j1:<https://www.w3.org/2006/time#> \r\n"
+                +"SELECT ?resource ?mediatype ?creationTime ?agent ?simulationTime ?scenario \r\n"
+                +"WHERE { \r\n"
+                +"OPTIONAL {?resource dcterms:format ?mediatype .}. \r\n"
+                +"OPTIONAL {?resource dcterms:created ?creationTime .}. \r\n"
+                +"OPTIONAL {?resource dcterms:creator ?agent .}. \r\n"
+                +"OPTIONAL {?resource j1:hasTime ?inst .}. \r\n"
+                +"OPTIONAL {?inst j1:inXSDDateTime ?simulationTime .}. \r\n"
+                +"FILTER ( regex(str(?resource), \"C://JPS_DATA/workingdir/JPS_SCENARIO\") ) \r\n"
+                +"OPTIONAL {?resource dcterms:isPartOf ?scenario .}. \r\n"
+                +"} ORDER BY DESC(?simulationTime) \r\n"
+                +"LIMIT 1000";
+        String metadataSetUrl= KeyValueManager.get(IKeys.URL_RDF_METADATA);
+        String expected1="expectedResult";
+        try(MockedStatic<MetaDataQuery> mtq = Mockito.mockStatic(MetaDataQuery.class)){
+            mtq.when(()->MetaDataQuery.query(sparql,metadataSetUrl)).thenReturn(expected1);
+            String actual= MetaDataQuery.query(sparql,metadataSetUrl);
+            Assert.assertEquals(expected1,actual);
+        }
+
+        String expected2="queryResultString";
+        try(MockedStatic<AccessAgentCaller> aac= Mockito.mockStatic(AccessAgentCaller.class)){
+            aac.when(()->AccessAgentCaller.query(MetaDataAnnotator.getMetadataSetUrl(),null,sparql)).thenReturn(expected2);
+            String actual=MetaDataQuery.query(sparql,"");
+            Assert.assertEquals(expected2,actual);
+        }
+    }
+
+    @Test
+        public void testQueryResources1(){
```

Comment:
Same as above, can add a note that testQueryResources1 is test case with 8 inputs, while testQueryResources2 is for test case with 3 inputs